docs/madr/decisions/030-extended-global-insights.md

```diff
@@ -0,0 +1,125 @@
+# Extended Global Insights
+
+* Status: accepted
+
+Technical Story: https://github.com/kumahq/kuma/issues/7707
+
+## Context and Problem Statement
+
+We would like to extend current Global Insight to add more meaningful metrics to it. We would like to add info about number of policies, zones,
+services, dataplanes and their status.
+
+## Considered Options
+
+* Extend `GlobalInsight` with data from `MeshInsight` and `ZoneInsight`.
+
+## Decision Outcome
+
+We will extend Global Insight with detailed information about number of policies, zones, service, dataplanes and its status. It will be done using 
+information from `MeshInsight` and `ZoneInsight`
+
+### Solution
+
+Right now Global Insight is really simple, it only returns count of resources, example response:
+
+```json
+{
+  "type": "GlobalInsights",
+  "creationTime": "2023-09-07T07:04:48.473293841Z",
+  "resources": {
+    "GlobalSecret": {
+      "total": 6
+    },
+    "Mesh": {
+      "total": 1
+    },
+    "Zone": {
+      "total": 0
+    },
+    "ZoneEgress": {
+      "total": 0
+    },
+    "ZoneIngress": {
+      "total": 0
+    }
+  }
+}
+```
+
+We would like to add more detailed information about zones, services and dataplanes. New extended Global Insight 
+should look like this:
+
+```json
+{
+   "services": {
+      "total": 5,
+      "internal": 4,
+      "external": 1,
+      "gatewayBuiltin": 1,
+      "gatewayProvided": 2,
+      "internalByStatus": {
+         "online": 2,
+         "offline": 1,
+         "partiallyDegraded": 1
+      }
+   },
+   "zones": {
+      "controlPlanes": {
+         "online": 1,
+         "total": 1
+      },
+      "zoneEgresses": {
+         "online": 1,
+         "total": 1
+      },
+      "zoneIngresses": {
+         "online": 1,
+         "total": 1
+      }
+   },
+   "dataplanes": {
+      "online": 23,
+      "offline": 10,
+      "partiallyDegraded": 17,
+      "total": 50
+   },
```

Comment:
Will we ever need these `dataplane` numbers split into sidecar/gateway? I suppose maybe when knowing health and specifically unhealthy its not that important what type they are just that they are unhealthy? Thought I'd ask in case. Ah, I also just noticed below here that we aren't specifically splitting policies by type either, so I guess this is the same.

Reply:
I would actually like those to be only sidecar dataplanes and not gateways. It’s not meaningful to have a combined count of DPPs and Gateways in my opinion.

Reply:
> I suppose maybe when knowing health and specifically unhealthy its not that important what type they are just that they are unhealthy? 

Yeah this ☝️ I suppose, thoughts on the policies?

Reply:
> I suppose maybe when knowing health and specifically unhealthy its not that important what type they are just that they are unhealthy?

It’s rather meaningless if you don’t know which type is unhealthy.

Reply:
Oh wait...

> I would actually like those to be only sidecar dataplanes and not gateways. It’s not meaningful to have a combined count of DPPs and Gateways in my opinion.

Ah ok so what you are saying is these shouldn't be/aren't a mix? You would like them to be sidecars only? Is it not useful to know that a gateway is unhealthy?

Reply:
It’s probably useful to know both, but currently (and that’s on `/mesh-insights`, not `/global-insights`), these counts are a combination of DPPs and Gateways because they’re the same resource underneath.

In the UI, we specifically have places where the counts for DPPs are shown and this won’t be correct with them being based on both DPPs and Gateways (i.e. DPPs are more important for UI purposes currently).

Reply:
Ok so from what you just said, it sounds like in an ideal world it would be useful if these were split?

> It’s probably useful to know both

> we specifically have places where the counts for DPPs are shown and this won’t be correct with them being based on both DPPs and Gateways

i.e. useful to know both _and_ we need to know how many of only only type in the UI _currently_

I get if there is a technical reason not to do this now, but in an ideal world would we/might we want to know these total split by type at some point?

Reply:
Does the same reasoning apply to policies? I know specifically of a places in the UI were we split the totals of these.

Reply:
We might do this in a mesh-specific context (where we use `/mesh-insights`), but on a global level, I don’t think we need this split. The total number of policies should be good enough.

Reply:
This is fine for me in the current state now, so if there’s nothing more, feel free to resolve.